packages/twenty-server/src/engine/core-modules/file/resolvers/file.resolver.ts

```diff
@@ -0,0 +1,54 @@
+import { UseFilters, UseGuards, UsePipes } from '@nestjs/common';
+import { Args, Mutation, Resolver } from '@nestjs/graphql';
+
+import { FileDTO } from 'src/engine/core-modules/file/dtos/file.dto';
+import { FileMetadataService } from 'src/engine/core-modules/file/services/file-metadata.service';
+import { PreventNestToAutoLogGraphqlErrorsFilter } from 'src/engine/core-modules/graphql/filters/prevent-nest-to-auto-log-graphql-errors.filter';
+import { ResolverValidationPipe } from 'src/engine/core-modules/graphql/pipes/resolver-validation.pipe';
+import { Workspace } from 'src/engine/core-modules/workspace/workspace.entity';
+import { AuthWorkspace } from 'src/engine/decorators/auth/auth-workspace.decorator';
+import { WorkspaceAuthGuard } from 'src/engine/guards/workspace-auth.guard';
+
+@UseGuards(WorkspaceAuthGuard)
+@UsePipes(ResolverValidationPipe)
+@UseFilters(PreventNestToAutoLogGraphqlErrorsFilter)
+@Resolver()
+export class FileResolver {
+  constructor(private readonly fileMetadataService: FileMetadataService) {}
+
+  @Mutation(() => FileDTO)
+  async createFile(
+    @AuthWorkspace() { id: workspaceId }: Workspace,
+    @Args('name') name: string,
+    @Args('fullPath') fullPath: string,
+    @Args('size') size: number,
+    @Args('type') type: string,
```

Comment:
I would have expected fullPath and size to be computed on the server size?  Could be abused otherwise no?

Reply:
Then we should just pass the id in that case.
I think in our case we can create the attachment directly during the initial upload. 
The pattern of doing it in two times makes sense (create the upload, then create the relationship). But since you introduced "file" as a metadata table then it makes sense to consider that all file uploaded will be stored in the metadata table (which doesn't mean we can't link it into different places with sub-sequent requests)

Reply:
Happy to discuss that/clarify on Discord if it does not make sense!